src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -406,7 +416,10 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                     parameter,
                     value: null);
 
-                arguments[i] = result.IsModelSet ? result.Model : parameter.DefaultValue;
+                if (result.IsModelSet)
+                {
+                    arguments[i] = result.Model;
+                }
```

Comment:
I think this is causing tests to blow up. I'm going to hardcode what we had before here even though it's not 100% right and log a bug for the follow up.